MMCore/MMCore.cpp

```diff
@@ -6700,6 +6700,34 @@ void CMMCore::saveSystemConfiguration(const char* fileName) throw (CMMError)
    {
       os << MM::g_CFGCommand_Property << ',' << MM::g_Keyword_CoreDevice << ',' << MM::g_Keyword_CoreFocus << ',' << focus->GetLabel() << endl;
    }
+
+   // save pixel size configurations
+   os << "# PixelSize settings" << endl;
+
+   vector<string> configs = getAvailablePixelSizeConfigs();
+   // normal group records
+   for (size_t j=0; j<configs.size(); j++)
+   {
+      Configuration c = getPixelSizeConfigData(configs[j].c_str());
+      for (size_t k=0; k<c.size(); k++)
+      {
+         PropertySetting s = c.getSetting(k);
+         os << MM::g_CFGCommand_ConfigPixelSize << ',' << configs[j] 
+            << ',' << s.getDeviceLabel() << ',' << s.getPropertyName() << ',' << s.getPropertyValue() << endl;
+      }
+      
+      os << MM::g_CFGCommand_PixelSize_um << ',' << configs[j] << ',' << getPixelSizeUmByID(configs[j].c_str()) << endl;
+
+      std::vector<double> affine = getPixelSizeAffine(configs[j].c_str());
```

Comment:
This is a C++ trap. `getPixelSizeAffine(bool)` is being called (with the non-null pointer interpreted as `true`); there is no `getPixelSizeAffine(const char *)`.

Reply:
Note that it already compiles, just returns the wrong data. I think you can use `getPixelSizeAffineByID(configs[j].c_str())`.

Reply:
thanks!